app/ui/src/main/java/com/fsck/k9/activity/MessageList.java

```diff
@@ -1698,7 +1698,9 @@ private void initializeFromLocalSearch(LocalSearch search) {
 
         List<String> folderServerIds = search.getFolderServerIds();
         singleFolderMode = singleAccountMode && folderServerIds.size() == 1;
-        if (singleFolderMode) {
+        if (drawer == null) {
+            return;
```

Comment:
This skips the last line of the method that seems unrelated to the drawer. My suggestion is to extract the drawer-related code to a separate method and do the `null` check there.

Reply:
I thought it was related, but you are right.

Reply:
> My suggestion is to extract the drawer-related code to a separate method and do the null check there.

That little method would need two parameters. Doesn't make sense here, I'd say.